src/components/polarity-panel.tsx

```diff
@@ -41,22 +41,51 @@ export class PolarityPanelComponent extends BaseComponent<IProps, IState> {
   private renderBarPolarityPanel = () => {
     const {simulation} = this.stores;
     const mag = simulation.getMagnetAtIndex(this.props.index);
+    const magType: MagnetType | null = mag ? mag.type : null;
+    const magRight = simulation.getMagnetAtIndex(1);
     const polaritylabel = mag && mag.barPolarity ? mag.barPolarity : "N-S";
     const polarityOn = polaritylabel === "S-N" ? true : false;
+    let posClass = "center";
+    if (this.props.index === 0 && magRight) {
+      posClass = "left";
+    } else if (this.props.index === 1) {
+      posClass = "right";
+    }
+    posClass = posClass + "-" + magType;
     return (
-      <SwitchPanelComponent
-        title="Polarity"
-        switchOn={polarityOn}
-        buttonText={polaritylabel}
-        buttonClick={this.handleClickPolarityButton}
-      />
+      <div className={"polarity-panel " + posClass}>
+        <svg className="icon bar-magnet-polarity-back2">
+          <use xlinkHref="#icon-bar-magnet-polarity-back2"/>
+        </svg>
+        <svg className="icon bar-magnet-polarity-back1">
+          <use xlinkHref="#icon-bar-magnet-polarity-back1"/>
+        </svg>
+        <div className="vertical-container">
+          <div className="title">Polarity</div>
+          <SwitchComponent
+            switchOn={polarityOn}
+            label={polaritylabel}
+            alwaysEnabled={true}
+            buttonClick={this.handleClickPolarityButton}
+          />
+        </div>
+      </div>
     );
   }
 
   private renderCoilPolarityPanel = () => {
     const {simulation} = this.stores;
     const mag = simulation.getMagnetAtIndex(this.props.index);
+    const magType: MagnetType | null = mag ? mag.type : null;
+    const magRight = simulation.getMagnetAtIndex(1);
     const magCoilPolarityVal = mag != null ? mag.coilPolarity : "plus-minus";
+    let posClass = "center";
+    if (this.props.index === 0 && magRight) {
+      posClass = "left";
+    } else if (this.props.index === 1) {
+      posClass = "right";
+    }
+    posClass = posClass + "-" + magType;
```

Comment:
Probably some way to clean up the redundancy within this file + in the strength panel, but not a big deal

Reply:
Yeah, there is definitely some redundancy.  If I get another chance to comb through the CSS or these React components again, I'll definitely take a stab at it.